pkg/proxy/proxier.go

```diff
@@ -319,38 +319,43 @@ type Proxier struct {
 	hostIP        net.IP
 }
 
+var (
+	ErrProxyOnLocalhost = fmt.Errorf("cannot proxy on localhost")
```

Comment:
Typed errors are fine (though it's not my favorite aspect of Go, whatevs).
My point was to grok why some errors matter more than others.

On Fri, May 15, 2015 at 7:32 AM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/proxy/proxier.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8269#discussion_r30412964>
> :
>
> > @@ -319,38 +319,43 @@ type Proxier struct {
> >  	hostIP        net.IP
> >  }
> >
> > +var (
> > +	ErrProxyOnLocalhost = fmt.Errorf("cannot proxy on localhost")
>
> So then we have to expose a method to check it. Any everyone has to check
> it the same way. And everyone needs a consistent message. Seems like you're
> arguing against people returning known error types from functions? If I set
> up a TLS config and get back a TLS typed error (this is invalid) isn't that
> reasonable?
> … <#14d57fde9d752fa5_>
> ----- Original Message -----
>  > @@ -319,38 +319,43 @@ type Proxier struct { > hostIP net.IP > } > >
> +var ( > + ErrProxyOnLocalhost = fmt.Errorf("cannot proxy on localhost")
> Shouldn't you pre-validate that, then, rather than surfacing it from the
> depths of the app? It's a pretty specific error to catch, you might as well
> check it yourself. --- Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/8269/files#r30387927
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/8269/files#r30412964>
> .
>